### ISSUE TITLE
[RFC] support customized search query tool

### ISSUE THREAD

**Issue by zhichao-aws** [opensearch-project/skills#318]
**Is your feature request related to a problem?**
In current implementation we only supports neural sparse tool and vectorDB tool, which build the search request by filling a fixed template. However, we have different use cases which can have complexed search queries. For example, search with conditions/filters, using hybrid query, using nested search(with chunking processor) etc

**What solution would you like?**
Implement a customized search tool, user can specify a search request template. The template accepts one parameter of search text. And RAG tool should also supports this customized search tool.

**Comment by dblock** [comment]
Thanks for opening this feature request.

[Catch All Triage - Attendees [1](https://github.com/dblock/), [2](https://github.com/krisfreedain), [3](https://github.com/andrross), [4](https://github.com/gkamat), [5](https://github.com/getsaurabh02)]

**Comment by zane-neo** [comment]
This tool accepts a new parameter: template and the search text will be filled into the template and then the template is the request body, is this understanding correct? Is it possible that user wants not only the query text as input parameter?

**Comment by zhichao-aws** [comment]
> This tool accepts a new parameter: template and the search text will be filled into the template and then the template is the request body, is this understanding correct? Is it possible that user wants not only the query text as input parameter?

Yes this is the correct understanding. My initial thought was users may need to use complexed filters or search DSLs instead of a naive neural or neural_sparse query, so I created this RFC to collect feedbacks. But now it seems no user valiate this use case. We can hold this untill recieve feedbacks from users
